azurerm/resource_arm_virtual_machine.go

```diff
@@ -957,7 +979,27 @@ func flattenAzureRmVirtualMachineOsProfileSecrets(secrets *[]compute.VaultSecret
 	return result
 }
 
-func flattenAzureRmVirtualMachineDataDisk(disks *[]compute.DataDisk) interface{} {
+func getStoppedVMManagedDiskInfo(idStr string, meta interface{}) (*compute.Disk, error) {
+	client := meta.(*ArmClient).diskClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(idStr)
+	if err != nil {
+		return nil, err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["disks"]
+
+	resp, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		return nil, fmt.Errorf("[ERROR] Error making Read request on stopped VM Azure Managed Disk %s (resource group %s): %s", name, resGroup, err)
+	}
+
+	return &resp, nil
+}
+
+func flattenAzureRmVirtualMachineDataDisk(disks *[]compute.DataDisk, stopped bool, meta interface{}) (interface{}, error) {
```

Comment:
This function is taking more responsibility, we may make it simpler and only do flatten work.